dash.el

```diff
@@ -1170,20 +1170,28 @@ returns the header value, but only after seeing at least one
 other value (the body)."
   (--partition-by-header (funcall fn it) list))
 
-(defun -partition-after-pred (pred list)
+(defmacro -partition-after-pred (pred list)
   "Partition directly after each time PRED is true on an element of LIST."
-  (when list
-    (let ((rest (-partition-after-pred pred
-                                       (cdr list))))
-      (if (funcall pred (car list))
-          ;;split after (car list)
-          (cons (list (car list))
-                rest)
-
-        ;;don't split after (car list)
-        (cons (cons (car list)
-                    (car rest))
-              (cdr rest))))))
+  (declare (debug (pred pred)))
+  (let ((r (make-symbol "result"))
+        (s (make-symbol "sublist"))
+        (l (make-symbol "list")))
+    `(let ((,l ,list))
+       (when ,l
+         (let* ((,r nil)
+		(,s nil))
```

Comment:
Please consistently indent with spaces, in accordance with the `indent-tabs-mode` setting in the project's `.dir-locals.el` file.

Reply:
Fixed, good catch.